### PR TITLE
Create ecosystem service credit spec

### PR DESCRIPTION
**Pull request by aaronc**
This PR creates a specification for ecosystem service credits - a form of non-fungible token (NFT) that has uniqueness constraints around geolocation, time, and credit type to prevent double issuance of credits for the same location and the same time.

Let us use this PR as a forum to discuss the design of this functionality.

### CODE REVIEW

File: docs/specs/ecosystem-service-credit.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+# Messages
+
+## Credit Issuance, Transfer and Consumption
```
Comment (ethanfrey) [inline docs/specs/ecosystem-service-credit.md]: This first section seems clear enough.
Mainly notes that make sense after our conversations.

If this is a spec for me to implement, fine. If this is a spec to post to explain the design to others, this needs much more English and not just the code structs

File: docs/specs/ecosystem-service-credit.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+// credit has been consumed, it can no longer be transferred
+type MsgConsumeCredit struct {
+  Credit CreditID
+  Holder sdk.AccAddress
+  Units sdk.Dec
+}
+```
+
+
+## Credit Exchange
```
Comment (ethanfrey) [inline docs/specs/ecosystem-service-credit.md]: I think the Exchange is a whole different topic and I would place that elsewhere.

Send and Consume is what we need now
Trying to design a marketplace / orderbook / dex takes a bit more design work.
I did start one simple on-chain orderbook here for example: https://github.com/iov-one/tutorial/
And this is a super rough sketch of behavior: https://github.com/iov-one/tutorial/blob/master/x/orderbook/README.md

Please break this section out to another PR.

**Comment by aaronc** [comment]
@ethanfrey I have removed the DEX specification from this spec. Also, not sure if you saw it but the English language spec I have been working on with Ron and Daniel is here: https://www.notion.so/regennetwork/Ecosystem-Service-Credit-Module-Specification-07399feee49c428dbfaf03b9a1529100

**Comment by ethanfrey** [comment]
I think it would be great to get clarity on the human-language version, based on product and market needs. And then revisit this.

To make sure the requirements and use cases are well spec'd before we work out the implementation details.

**Comment by clevinson** [comment]
Closing as a duplicate of a newer ticket.

**Pull request closed without merging**